sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/AggUtils.scala

```diff
@@ -113,6 +114,9 @@ object AggUtils {
         resultExpressions = partialResultExpressions,
         child = child)
 
+    val interExec: SparkPlan = mayAppendMergingSessionExec(groupingExpressions,
```

Comment:
Could you add a comment here explaining why we need to append `MergingSessionExec`?